DOC-1871: RC - SAML Integration guide migration (#2345)
RedisLabs/redislabs-docs: content/rc/security/single-sign-on/saml-sso/saml-integration-auth0.md

---
Title: Auth0 SAML integration guide
linkTitle: Auth0 integration
description: This integration guide shows how to configure Auth0 as a SAML single sign on provider for your Redis Cloud account.
weight: 58
alwaysopen: false
categories: ["RC"]
---

This guide shows how to configure [Auth0](https://auth0.com/docs) as a SAML single sign-on identity provider (IdP) for your Redis Cloud account.

To learn more about Redis Cloud support for SAML, see [SAML single sign-on]({{<relref "/rc/security/single-sign-on/saml-sso">}}).

## Step 1: Set up your identity provider (IdP)

### Specify the SAML owner

1. Sign in to your Auth0 account and navigate to **User Management > Users**.
   * Select the SAML owner.
   * Verify the details.

    SAML assertion requires first and last name, which are not available in the default user profile. 

>     > **Note**: Depending how they are created, users can have different profiles. See Sample User Profiles.

Reviewer: ```suggestion
    > **Note**: Depending how they are created, users can have different profiles. 
```

---
Title: Auth0 SAML integration guide
linkTitle: Auth0 integration
description: This integration guide shows how to configure Auth0 as a SAML single sign on provider for your Redis Cloud account.
weight: 58
alwaysopen: false
categories: ["RC"]
---

This guide shows how to configure [Auth0](https://auth0.com/docs) as a SAML single sign on identity provider (IdP) for your Redis Cloud account.

To learn more about Redis Cloud support for SAML, see [SAML single sign on]({{<relref "/rc/security/single-sign-on/saml-sso">}}).

## Step 1 - Setup your identity provider (IdP)

### Prepare the SM "owner" user

1. Sign in in your Auth0 account and open **User Management → Users**
   * Select the User that will be SAML owner in SM
   * Verify the details

    First & Last names are needed for the SAML assertion and are not available in the default user profile. 

    > **Note**: Depending how the users are created, these can have different profiles, see: Sample User Profiles

    {{<image filename="images/rc/saml/auth0_saml_1.png" alt="" >}}{{</image>}}

> 2. Add **user_metadata** to fulfill the SAML assertion then click the **Save** button. The key-value pair of **redisAccountMapping** consists of the **lower-cased role name** (ie owner, member, manager, or viewer) AND your **Redis Cloud Account ID** (you can find this information in the upper-right user menu at app.redislabs.com).

Reviewer 2: ```suggestion
2. Add **user_metadata** to fulfill the SAML assertion then click the **Save** button. The key-value pair of **redisAccountMapping** consists of the **lower-cased role name** (owner, member, manager, or viewer) AND your **Redis Cloud Account ID** found in the [account settings]({{<relref "rc/accounts/account-settings">}}).
```

We usually don't use directional language such as upper-right, so adding a reference to where you can find the account ID.

      "nameIdentifierProbes": [
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
      ]
    }
    ```

    Scroll down and **Save** the configuration

### IdP initiated SSO

To use IdP-initiated SSO with certain identity providers, you also need to set the RelayState parameter to the following URL:

`https://app.redislabs.com/#/login/?idpId=<ID>`

> **Note**: Replace <ID> so it matches the AssertionConsumerService Location URL’s ID (the content after the last forward slash "/"). To learn more about how to configure service provider apps, see your identity provider’s documentation.

## Step 4 - Return to Redis Cloud SM

1. Return to Redis Cloud SM, and now click on the **Activate** button

    {{<image filename="images/rc/saml/sm_saml_8.png" alt="" >}}{{</image>}}

2. A popup will appear, explaining that in order to test the SAML connection, that we need to login with credentials of a user defined in Auth0.

    {{<image filename="images/rc/saml/sm_saml_9.png" alt="" >}}{{</image>}}

3. The Auth0 login screen will appear, enter the credentials and click **Sign In**

    {{<image filename="images/rc/saml/auth0_saml_12.png" alt="" >}}{{</image>}}

> 4. If the test has succeeded, you will see the following screen. Your local account is now considered a SAML account. In order to login to SM going forward, you click on the **Sign in with SSO** button.

Reviewer 3: Maybe change "going forward" to "in the future".
